sdk/servicebus/service-bus/src/receiver.ts

```diff
@@ -315,10 +322,11 @@ export class SessionReceiver {
    */
   async renewLock(): Promise<Date> {
     this._throwIfReceiverOrConnectionClosed();
-    this._messageSession.sessionLockedUntilUtc = await this._context.managementClient!.renewSessionLock(
+    await this._createMessageSessionIfDoesntExist();
```

Comment:
We don't need to check `this.sessionId` here?

Reply:
The checking for session is what seems strange to me - because ifDoesn'tExist implies it'll be a no-op if it has nothing to do.

Reply:
Here is what I was thinking (which I should have added in the comments)
- Any time you see `_messageSession` it refers to an instance of the class MessageSession which when instantiated creates an AMQP receiver link
- Management operations like peek and receiveDeferredMessages do not need an active AMQP link. All they need is a sessionId. If the sessionId was invalid the operation would fail with an appropriate error message. If sessionId was not provided, then we need the AMQP link so that Service Bus can pick a sessionId for us.
- Other management operations like renewLock, setState, getState require the AMQP link to exist

Based on the above, the `this.sessionId` check is added before calling `_createMessageSessionIfDoesntExist` to avoid creating an AMQP link when not required.

Reply:
I'll write tests to ensure that peek and receiveDeferredMessages will work without the AMQP receiver link

Reply:
Tests for peek done.
Existing tests for receive deferred message will cover the above case